lib/mix/lib/mix/tasks/test.ex

```diff
@@ -210,7 +210,8 @@ defmodule Mix.Tasks.Test do
 
       mix test --only external
 
-  Which is equivalent to:
+  If no tests match and the `--only` option was used the test suite will fail.
```

Comment:
How about rephrasing it a bit like this:
If no tests are matched and the `--only` option has been provided, the test suite will fail.

Reply:
Both suggestions are way better than what I wrote. I'm going to use @fertapric's